mcp/mastra/verticals/routing/agent.ts

```diff
@@ -0,0 +1,80 @@
+import type { Agent } from '@mastra/core/agent';
+import type { Tool } from '@mastra/core/tools';
+import { keyBy } from 'lodash-es';
+import { createAgent } from '../../utils/index.js';
+import { getPublicAgents } from '../index.js';
+import { routingTools } from './tools.js';
+
+/**
+ * Routing Agent Configuration Options
+ */
+export interface RoutingAgentOptions {
+  /**
+   * Override the default tools with custom tools (useful for testing)
+   */
+  tools?: Record<string, Tool>;
```

Comment:
Tests should also be able to override agents, setting it to empty. That's important during testing.